### PR TITLE
[COASTAL-674] record alert that is issued and retracted at the same time

### PR DESCRIPTION
**Pull request by nhamming**
https://tidepool.atlassian.net/browse/COASTAL-674

### CODE REVIEW

**Review by rickpasetto — APPROVED** [review]
LGTM!  (Name sounds find to me too! 👍 )

**Comment by nhamming** [comment]
added the ability to detect if an persisted alert already exists

File: LoopKit/Alert.swift [hunk 1]
```diff
@@ -35,15 +35,21 @@ public struct PersistedAlert: Equatable {
     }
 }
 
-/// Protocol for looking up alerts persisted in storage
+/// Protocol for recording and looking up alerts persisted in storage
 public protocol PersistedAlertStore {
+    /// Determine if an alert is already recorded for a given `Alert.Identifier`.
```
Comment (rickpasetto) [inline LoopKit/Alert.swift]: Any alert, regardless of retraction?  Also, this is already `PersistedAlertStore`, so no need IMO to repeat `Persisted`.  Consider naming `doesIssuedAlertExist` instead (or `doesUnretractedAlertExist` or whatever condition this API specifies).  Also consider updating the comment.

**Review by rickpasetto — COMMENTED** [review]
Added a comment...

File: LoopKit/Alert.swift (hunk 1 above)
Comment (nhamming) [reply LoopKit/Alert.swift]: This is intended to find an alert irrelevant of its state ([un]retracted | [un]acknowledged). Just that it exists in the store.

**Pull request merged**